docroot/modules/custom/sitenow_intranet/sitenow_intranet.module

```diff
@@ -10,97 +10,91 @@
+        unset($variables['content']);
+        break;
 
-  $page['main']['title_container'] = [
-    '#type' => 'html_tag',
-    '#tag' => 'div',
-    '#attributes' => [
-      'class' => [
-        'section-padding__top-extra',
-        'layout--title',
-        'layout__container',
-        'layout',
-      ],
-    ],
-  ];
-  $page['main']['title_container']['title'] = [
-    '#prefix' => '<h1 class="page-title headline--serif headline text-align-center">',
-    '#suffix' => '</h1>',
-  ];
+      case 'system_main_block':
+        $config = \Drupal::config('sitenow_intranet.settings');
+        $status_code_map = IntranetHelper::getStatusCodeMap();
 
-  // Display a different message to authenticated users with no other roles.
-  $roles = \Drupal::currentUser()->getRoles();
-  if (count($roles) === 1 && in_array('authenticated', $roles)) {
-    $page['main']['title_container']['title']['#markup'] = $config->get('unauthorized.title');
+        $variables['content'] = [
+          '#markup' => check_markup($config->get("{$status_code_map[$code]}.message"), 'minimal'),
+          '#prefix' => '<div class="text-align-center">',
+          '#suffix' => '</div>',
+        ];
 
-    $page['main']['message'] = [
-      '#markup' => check_markup($config->get('unauthorized.message'), 'minimal'),
-      '#prefix' => '<div class="text-align-center">',
-      '#suffix' => '</div>',
-    ];
+        // If the user is logged-in, show a logout link on the access denied
+        // page.
+        if ($code === 403 && \Drupal::currentUser()->isAuthenticated()) {
+          $variables['content']['logout_link'] = [
+            '#type' => 'link',
+            '#title' => t('Logout'),
+            '#url' => Url::fromRoute('user.logout') ,
+            '#prefix' => '<p class="text-align-center">',
+            '#suffix' => '</p>',
+            '#attributes' => [
+              'class' => [
+                'bttn',
+                'bttn--primary',
+                'bttn--caps',
+              ],
+            ],
+          ];
+        }
+        break;
 
-    // If the user is authenticated but has no other roles, return a 403.
-    $page['#attached']['http_header'][] = ['Status', '403'];
+    }
   }
-  else {
-    $page['main']['title_container']['title']['#markup'] = $config->get('access_denied.title');
+}
 
-    $page['main']['message'] = [
-      '#markup' => check_markup($config->get('access_denied.message'), 'minimal'),
-      '#prefix' => '<div class="text-align-center">',
-      '#suffix' => '</div>',
-    ];
+/**
+ * Implements hook_preprocess_page().
+ */
+function sitenow_intranet_preprocess_page(&$variables) {
+  $code = IntranetHelper::getStatusCode();
 
-    // If the user is not authenticated, return a 401.
-    $page['#attached']['http_header'][] = ['Status', '401'];
+  if (IntranetHelper::checkStatusCode($code)) {
+    foreach ([
+      'action_menu',
+      'footer_first',
+      'footer_second',
+      'primary_menu',
```

Comment:
```suggestion
      'pre_footer',
      'primary_menu',
      'region_content',
```

what about alerts? hawk alerts and custom alerts are showing

Reply:
Good question. We're kind of getting into new territory here, since this is locking down more than what the old system locked down. If we are going to remove other things from being displayed, then we probably want to do that as a follow up so we can have some discussion around it?

Removing `action_menu` will cause our test to break. We may want to consider updating the test to place the menu in `primary_menu`? Or add a separate test to check that?

Reply:
oh. i didn't intend to remove action_menu. i think github is highlighting the area wrong. just adding prefooter alphabetically, keeping primary menu and adding region_content (after content). These regions were added after Intranet was made. Alerts was a missed region the first pass. "an important announcement (custom alert) to the intranet viewers about not talking to the press about xyz would be visible"